string subteststring
string[] subtestoptstring
string[] subteststringarray

int32 subtestint
int32[] subtestoptint
int32[] subtestintarray

float32 subtestfloat
float32[] subtestoptfloat
float32[] subtestfloatarray
